runtime/src/bank.rs

```diff
@@ -6829,6 +6829,9 @@ impl Bank {
             Arc::new(reserved_keys)
         };
 
+        // Update the cost-tracker's block limits.
+        self.update_block_limits();
```

Comment:
Whenever we create a completely new bank (from snapshot or genesis) this will set the limits according to the feature-set.

This is also called when we move to a new epoch via `new_from_parent`.